### DISCUSSION TITLE
Find most used colour / range of colours (alpha channel if present?) in a picture

### DISCUSSION THREAD

**Discussion by axlgit** [libvips/libvips#2154] [Q&A]
I cannot find anymore a question that I think I found in this discussions' section posted by somebody who wanted to whiten the noise background (light gray/white) on an scanned picture.
I remember that person asked for it because he claimed it would help him during the OCR process.
In my case I would need to find the most used colour or the biggest area interested by a range of colours centered around the one I specify and replace it by another colour. I admit I'm not very familiar with the topic so I would say that I may provide an interval for each of the RGBA values. As an example you may think of a scanned text page where the paper could be white or light blue or light red (with noise) and of course the text would be visible and readable.

Is there any function in pyvips? Or how may I proceed using pyvips' tools?
As said an input either by using an exact colour or by specifying a interval of RGB centered in the chosen colour will be good.

Thanks,

Alex

**Comment by jcupitt** [comment]
Hi, libvips has 3D histograms:

https://libvips.github.io/libvips/API/current/libvips-arithmetic.html#vips-hist-find-ndim

So an easy way is to make (perhaps) a 16x16x16 histogram and find the position of the maximum.
